tsconfig.json

```
@@ -2,7 +2,8 @@
   "extends": "@tsconfig/node16/tsconfig.json",
```

i know unrelated to this pr, but could pos simplify further extending our base instead?

```suggestion
  "extends": "./config/tsconfig.base.json",
```

so no longer extending `@tsconfig/node16/tsconfig.json` directly in two places

---

or can get this merged first and i'll follow-up

---

Yeah, maybe we should do this.  I wasn't really sure.  AFAICT, the workspace root configuration will be used in VS Code when a file should be typechecked (e.g., with a `// @ts-check` directive) but is otherwise not included in a `tsconfig.json`.  Think something like `scripts/foo.js` where the file should be typechecked, but it isn't shipped in the tarball nor are declarations generated for it.  But I'm not entirely sure--it might just use the closest ancestor `tsconfig.json`.   

The `composite: true` `compilerOption` in `tsconfig.base.json` doesn't make sense for the root workspace, either.  I don't think that has any negative consequences, but again, I'm not sure. It would be allowed to be referenced in as part of an incremental build setup in some `tsconfig.json` which is _not_ part of this repo?  That _might_ be useful, for somebody, somehow??

---

> Yeah, maybe we should do this. I wasn't really sure. AFAICT, the workspace root configuration will be used in VS Code when a file should be typechecked (e.g., with a `// @ts-check` directive) but is otherwise not included in a `tsconfig.json`. Think something like `scripts/foo.js` where the file should be typechecked, but it isn't shipped in the tarball nor are declarations generated for it. But I'm not entirely sure--it might just use the closest ancestor `tsconfig.json`.

yeah initially hard to tell 😅 briefly tested our current setup to shed some light
on `packages/aa/test/index.spec.js` adding `// @ts-check` and `const a = 5`
then adding to relevant tsconfig's `"compilerOptions": { "noUnusedLocals": true }`
- `tsconfig.json` - no effect
- `config/tsconfig.base.json` - TS6133
  - `config/tsconfig.test.json` - TS6133
    - `packages/aa/test/tsconfig.json` - TS6133
  - `config/tsconfig.workspace.json` - no effect
    - `packages/aa/tsconfig.json` - no effect

so looks like root `tsconfig.json` ignored
and yeah closest ancestor used

---

I think I'd prefer to leave it as-is.  My _intent_ for `tsconfig.base.json` is that _non-root_ `tsconfig.json` files extend it.  Since I'd rather not add comments to `tsconfig.base.json` (while allowed by TS, but possibly impedes integration with other tools), I'll create `config/README.md` explaining myself.